torch/_subclasses/meta_utils.py

```diff
@@ -94,7 +94,10 @@ def set_tensor_memo(self, t, v):
         # hold a weak ref to self, otherwise it will be kept alive
         # by the del_ten closure
         self_weak_ref = weakref.ref(self)
-        weak_st = StorageWeakRef(t.storage())
+        if t.is_sparse:
```

Comment:
Is it worth holding onto either the values or indices of the sparse tensor storage ?

Reply:
So, the sparse tensors are not supposed to alias with each other internally, so hypothetically this is not necessary. However, it is possible to get an aliasing reference to values and is often used by users to do an inplace modification to a sparse tensor. This needs to be worked out how to do correctly...

Reply:
Maybe add a `# TODO: sparse aliasing` here ?